app/models/dataset/Dataset.scala

```diff
@@ -291,13 +291,15 @@ class DatasetDAO @Inject()(sqlClient: SqlClient, datasetLayerDAO: DatasetLayerDA
             isUnreported = row._10 == unreportedStatus,
         ))
 
-  private def buildSelectionPredicates(isActiveOpt: Option[Boolean],
-                                       isUnreported: Option[Boolean],
-                                       organizationIdOpt: Option[ObjectId],
-                                       folderIdOpt: Option[ObjectId],
-                                       uploaderIdOpt: Option[ObjectId],
-                                       searchQuery: Option[String],
-                                       includeSubfolders: Boolean)(implicit ctx: DBAccessContext): Fox[SqlToken] =
+  private def buildSelectionPredicates(
+      isActiveOpt: Option[Boolean],
+      isUnreported: Option[Boolean],
+      organizationIdOpt: Option[ObjectId],
+      folderIdOpt: Option[ObjectId],
+      uploaderIdOpt: Option[ObjectId],
+      searchQuery: Option[String],
+      includeSubfolders: Boolean,
+      excludeRemovedOnDisk: Boolean = true)(implicit ctx: DBAccessContext): Fox[SqlToken] =
```

Comment:
This means the datasets marked as deleted are not included in the list query at all? I’d say they should still be included, just like the "no longer available on datastore" ones. I think there are already filters used by the frontend for active/inactive datasets. My guess is that the isUsable column is used for that. Also could you have another look at the `deactivateUnreported` logic in DatasetService? I’d guess that this will somehow interact. the deletedByUserStatus should probably be included in `inactiveStatusList`?

Reply:
I think I implemented this now, but I noticed that the status "Deleted by user" is actually never (or only for a short amount of time) present in the DB, because I trigger the rechecking of the inbox which checks the directories and the dataset is actually deleted on disk, so it will be marked as "No longer available on datastore.". Is this sufficient or should there be a more explicit status? Otherwise I could delete the "Deleted by user" status again.

Reply:
Hmm, I see. I think it would be valuable to preserve the information that it was in fact deleted by the user. Could you adapt the existing functions so that "Deleted by user" is not overwritten by "No longer available on datastore"?

Also, I do not yet fully see why this added check inbox trigger is needed now. Wouldn’t its effects already have happened on the wk side (delete layers, set status, etc)? Maybe I’m missing something though. It’s not like the new trigger should hurt much (it does potentially take some seconds, though, depending on IO speed)

Reply:
Fixed it now. This behavior was caused by a typo in an SQL query, which meant the status was never actually assigned. This is fixed now. A problem that persists is that the deleted dataset is still displayed in the dataset list. This status should be probably filtered (via a modified request).

Reply:
Cool! Hmm, I would have expected that these datasets have isActive=false, so the current query would skip them together with the "unreported" ones.

Maybe this has something to do with the `isUsable` bool? Just noticed that. Looks somewhat redundant to be honest, but I haven’t looked into it that much. Maybe you need to change that value when updating the status

Reply:
Not sure that would do anything. On my instance the only query parameters are folderId and compact, there are datasets with isActive being returned and they are also shown in the list.

Reply:
@philippotto could you comment on this? How does the frontend decide which datasets to show depending on the selected filters? is it the isActive property?

![image](https://github.com/scalableminds/webknossos/assets/1390035/16c91627-48da-43e9-a246-6967231c1d8c)

Reply:
> @philippotto could you comment on this? How does the frontend decide which datasets to show depending on the selected filters? is it the isActive property?

It uses the `isUnreported` property of each dataset.

Reply:
I think deleted datasets should not be shown on any filter? That would at least be unexpected for me. So they should be marked with isUnreported?

Reply:
Not certain what the best UX here would be. I would have said isUnreported (meaning show them if the "missing" filter is active) is good. That is at least some explanation why the name is still reserved. But maybe you prefer something different? cc @philippotto

Reply:
> I would have said isUnreported (meaning show them if the "missing" filter is active) is good. That is at least some explanation why the name is still reserved.

I agree with that (for "deleted" datasets where annotations exist). To avoid any potential misunderstandings, let me write out how I'd imagine this overall:

- deleting a DS for which no annotation exists should completely delete the DS so that it cannot be found via the UI (also not via any filters)
- deleting a DS for which annotations exist, should tell the user "we deleted the dataset, but since annotations exist for it, the dataset will still be listed under "unreported". if you want to delete the dataset completely, do ...".
  - the perceived behavior would be equal to when the user deletes a dataset from disk.

Does this make sense?

Reply:
Sounds good to me. On "if you want to delete the dataset completely, do ..." – leider gibt es aktuell keinen user-exposed weg hierfür. Das wäre ein follow-up issue. Daher würd ich den Part in der Message erstmal weglassen :sweat_smile:

Reply:
Ok, fair :)

Reply:
Deleted by user is now also a cause for datasets to be unreported.